Tasks/frmCleanup.cs

```diff
@@ -178,99 +178,222 @@ private void btnCleanup_Click(object sender, EventArgs e)
+
+                    var cookies = (roamingappdata + "\\Mozilla\\Firefox\\Profiles\\");
+                    foreach (string direc in Directory.EnumerateDirectories(cookies))
                     {
-                        try
+                        if (direc.Contains("release") == true)
                         {
-                            var cookiefile = (direc + "\\cookies.sqlite");
-                            File.Delete(cookiefile);
-                            CleanupLogsLBox.Items.Add("Firefox cookies cleared!");
+                            try
+                            {
+                                var cookiefile = (direc + "\\places.sqlite");
+                                File.Delete(cookiefile);
+                                CleanupLogsLBox.Items.Add("Deleted Firefox History!");
 
-                        }
-                        catch
-                        {
+                            }
+                            catch (Exception exc)
+                            {
 
-                            CleanupLogsLBox.Items.Add("Error when trying to delete firefox cookies!");
+                                //do nothing
 
-                        }
+                            }
 
+                        }
                     }
+
+                }
+                catch (Exception exc)
+                {
+                    CleanupLogsLBox.Items.Add("Error when trying to delete Firefox History! \n" + exc);
                 }
+
+
+
+
             }
-            if (checkBox16.Checked)
+            if (checkBox17.Checked)
             {
-
-                var cookies = (roamingappdata + "\\Mozilla\\Firefox\\Profiles\\");
-                foreach (string direc in Directory.EnumerateDirectories(cookies))
+                try
                 {
-                    if (direc.Contains("release") == true)
+
+                    var profiles = (roamingappdata + "\\Mozilla\\Firefox\\Profiles\\");
+                    foreach (string direc in Directory.EnumerateDirectories(profiles))
                     {
-                        try
+                        if (direc.Contains("release") == true)
                         {
-                            var cookiefile = (direc + "\\places.sqlite");
-                            File.Delete(cookiefile);
-                            CleanupLogsLBox.Items.Add("Deleted Firefox History!");
+                            try
+                            {
+                                var extensionsfolder = (direc + "\\extensions\\");
+                                foreach (string file in Directory.EnumerateFiles(extensionsfolder))
+                                {
+                                    try
+                                    {
+                                        File.Delete(file);
+                                        CleanupLogsLBox.Items.Add("Firefox Extension File " + file + " Removed.");
```

Comment:
There should be a way to sort all extensions and maybe pick which ones you want to remove instead of removing all of them